### PR TITLE
Change RoslynSettingsFileSynchronizer to only save suppressed issues

### PR DESCRIPTION
**Pull request by bigfluffycookie**
fixes https://github.com/SonarSource/sonarlint-visualstudio/issues/3868

### CODE REVIEW

File: src/Roslyn.Suppressions/Roslyn.Suppressions.UnitTests/InProcess/RoslynSettingsFileSynchronizerTests.cs [hunk 1]
```diff
@@ -194,6 +194,46 @@ public async Task UpdateFileStorage_IssuesAreConvertedAndFiltered()
             actualSuppressions[1].RoslynRuleId.Should().Be("S222");
         }
 
+        [TestMethod]
```
Comment (bigfluffycookie) [inline src/Roslyn.Suppressions/Roslyn.Suppressions.UnitTests/InProcess/RoslynSettingsFileSynchronizerTests.cs:197]: This test is almost a complete copy from the test above. The only change is that it tests if suppressed issues get filtered

File: src/Roslyn.Suppressions/Roslyn.Suppressions.UnitTests/TestHelper.cs [hunk 2]
```diff
@@ -37,11 +37,13 @@ internal static class TestHelper
...
-    public static SonarQubeIssue CreateSonarQubeIssue(string ruleId = "any",
-            int? line = null,
-            string filePath = "filePath",
-            string hash = "hash") =>
-            new SonarQubeIssue(
+        public static SonarQubeIssue CreateSonarQubeIssue(string ruleId = "any",
+                int? line = null,
+                string filePath = "filePath",
+                string hash = "hash",
+                bool isSuppressed = true)
```
Comment (bigfluffycookie) [inline src/Roslyn.Suppressions/Roslyn.Suppressions.UnitTests/TestHelper.cs:44]: Made the default for isSuppressed true so it does not interfere with existing tests.

File: src/Roslyn.Suppressions/Roslyn.Suppressions/InProcess/RoslynSettingsFileSynchronizer.cs [hunk 3]
```diff
@@ -112,7 +112,7 @@ public async Task UpdateFileStorageAsync()
 
                 if (!string.IsNullOrEmpty(sonarProjectKey))
                 {
-                    var allSuppressedIssues = serverIssuesStore.Get();
+                    var allSuppressedIssues = serverIssuesStore.Get().Where(x => x.IsResolved).ToArray();
```
Comment (bigfluffycookie) [inline src/Roslyn.Suppressions/Roslyn.Suppressions/InProcess/RoslynSettingsFileSynchronizer.cs:115]: Still a bit unsure about linq syntax so if this is wrong lmk

Comment (duncanp-sonar) [reply src/Roslyn.Suppressions/Roslyn.Suppressions/InProcess/RoslynSettingsFileSynchronizer.cs:115]: The syntax is fine.

You could skip the `ToArray` here, though; `allSuppressedIssues` is used in another LINQ expression immediately below which also has a `ToArray`, so we probably don't need it here as well.

An alternative approach would be to add the check into the LINQ expression below i.e.

```
       var allIssues = serverIssuesStore.Get();

       var settings = new RoslynSettings
                    {
                        SonarProjectKey = sonarProjectKey,
                        Suppressions = allIssues
                                            .Where(x => x.IsSuppressed)        <----- CHECK ADDED HERE
                                            .Select(x => IssueConverter.Convert(x))
                                            .Where(x => x.RoslynLanguage != RoslynLanguage.Unknown && !string.IsNullOrEmpty(x.RoslynRuleId))
                                            .ToArray(),
                    };

```

(I would probably go with the second approach and put all of the LINQ conditions in one place).

**Review by duncanp-sonar — APPROVED** [review]
Looks good. One comment/suggestion about the LINQ expression.

**Pull request merged**